gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
To clarify, when I say "Carriage sequence will change (reverse)", I mean producers will send `TripDescriptors` with the elements in `carriage_descriptor` reversed. For example, using this scenario
![trips](https://user-images.githubusercontent.com/6913473/28562858-4286aa16-7167-11e7-9b82-80db9a48d2cb.jpg)
 as a reference, the `VehicleDescriptors` corresponding to a trip with `direction_id = 1` would be:
```
carriage_descriptor {
  label = 'A'
  carriage_sequence: 1
},
carriage_descriptor {
  label = 'B'
  carriage_sequence: 2
},
carriage_descriptor {
  label = 'C'
  carriage_sequence: 3
}
```
While `direction_id = 0` would be:
```
carriage_descriptor {
  label = 'C'
  carriage_sequence: 1
},
carriage_descriptor {
  label = 'B'
  carriage_sequence: 2
},
carriage_descriptor {
  label = 'A'
  carriage_sequence: 3
}
```
Note also producers can change the carriage_descriptor at any time in the trip. If carriages are added/removed at a stop, then producers simply add/remove carriage_descriptor elements.